bigtable/spark/src/main/scala/example/DataFrameDemo.scala

```diff
@@ -0,0 +1,92 @@
+package example
+
+import org.apache.hadoop.hbase.spark.datasources.{HBaseSparkConf, HBaseTableCatalog}
+
+import scala.util.Try
+
+object DataFrameDemo extends App {
+
+  println("Starting up...")
+
+  val projectId = args(0)
+  val instanceId = args(1)
+  val table = Try(args(2)).getOrElse("dataframe-demo")
+  val numRecords = Try(args(3).toInt).getOrElse(10)
+
+  import org.apache.spark.sql.SparkSession
+  val spark = SparkSession.builder().master("local[*]").getOrCreate()
+  println("Spark version: " + spark.version)
+
+  val catalog =
+    s"""{
+       |"table":{"namespace":"default", "name":"$table", "tableCoder":"PrimitiveType"},
+       |"rowkey":"key",
+       |"columns":{
+       |"col0":{"cf":"rowkey", "col":"key", "type":"string"},
+       |"col1":{"cf":"cf1", "col":"col1", "type":"boolean"},
+       |"col2":{"cf":"cf2", "col":"col2", "type":"double"},
+       |"col3":{"cf":"cf3", "col":"col3", "type":"int"}
+       |}
+       |}""".stripMargin
+  println(s"Writing $numRecords records to $table")
+
+  // FIXME Explain the options
```

Comment:
`HBaseTableCatalog.newTable -> "5",` needs an explanation before this PR can be merged
The other options can be left as a FIXME for now